lib/ansible/plugins/lookup/filetree.py

```diff
@@ -100,6 +100,22 @@ def file_props(root, path):
 
     return ret
 
+def walk(path, topdown=None):
+    topdown = topdown or False
+
+    ret = []
+    for root, dirs, files in os.walk(path, topdown=True):
+        for entry in dirs + files:
+            relpath = os.path.relpath(os.path.join(root, entry), path)
+
+            # Skip if relpath was already processed (from another root)
+            if relpath in [entry['path'] for entry in ret]:
```

Comment:
This is really inefficient, it creates a O(n) list at each step. Either have a list of paths in parallel to ret or fine a more efficient way to check the path is already included.